### PR TITLE
feature(swap): order values remain when switching between limit and market 

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-608](https://emurgo.atlassian.net/browse/YOMO-608)

[YOMO-608]: https://emurgo.atlassian.net/browse/YOMO-608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx:1]: Basic component. Further logic will be implemented here: https://emurgo.atlassian.net/browse/YOMO-609

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditMarketPrice.tsx
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditMarketPrice.tsx:1]: @banklesss check with product, market price can't be changed, in this case I'd rename it to "ShowMarketPrice", I think there is no use case where you can update the market price.

**Review by stackchain — COMMENTED** [review]
.

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import {EditMarketPrice} from './ShowMarketPrice'
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    padding: 16,
+  },
+})
+
+storiesOf('Swap Edit Market Price', module)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx]:
```suggestion
storiesOf('Swap Show Market Price', module)
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.tsx [hunk 2]
```diff
@@ -10,11 +10,7 @@ import {useTokenInfo} from '../../../../../yoroi-wallets/hooks'
 import {useStrings} from '../../../common/strings'
 import {useSwapTouched} from './TouchedContext'
 
-type Props = {
-  disabled?: boolean
-}
-
-export const EditMarketPrice = ({disabled = false}: Props) => {
+export const EditMarketPrice = () => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.tsx]:
```suggestion
export const ShowMarketPrice = () => {
```

**Comment by banklesss** [comment]
thank you @stackchain

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  .addDecorator((story) => (
+    <SelectedWalletProvider wallet={mocks.wallet}>
+      <View style={styles.container}>{story()}</View>
+    </SelectedWalletProvider>
+  ))
+  .add('Default', () => <EditMarketPrice />)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx]:
```suggestion
  .add('Default', () => <ShowMarketPrice />)
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import {storiesOf} from '@storybook/react-native'
+import React from 'react'
+import {StyleSheet, View} from 'react-native'
+
+import {SelectedWalletProvider} from '../../../../../SelectedWallet'
+import {mocks} from '../../../../../yoroi-wallets/mocks'
+import {EditMarketPrice} from './ShowMarketPrice'
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowMarketPrice.stories.tsx]:
```suggestion
import {ShowMarketPrice} from './ShowMarketPrice'
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 5]
```diff
@@ -13,10 +13,11 @@ import {ButtonGroup} from '../../../common/ButtonGroup/ButtonGroup'
 import {useNavigateTo} from '../../../common/navigation'
 import {useStrings} from '../../../common/strings'
 import {EditBuyAmount} from './EditBuyAmount/EditBuyAmount'
-import {EditMarketPrice} from './EditMarketPrice'
+import {EditLimitPrice} from './EditLimitPrice'
 import {ShowPoolActions} from './EditPool/ShowPoolActions'
 import {EditSellAmount} from './EditSellAmount/EditSellAmount'
 import {EditSlippage} from './EditSlippage/EditSlippage'
+import {EditMarketPrice} from './ShowMarketPrice'
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]:
```suggestion
import {ShowMarketPrice} from './ShowMarketPrice'
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 6]
```diff
@@ -100,7 +101,7 @@ export const CreateOrder = () => {
 
         <Spacer height={20} />
 
-        <EditMarketPrice disabled={createOrder.type === 'market'} />
+        {createOrder.type === 'market' ? <EditMarketPrice /> : <EditLimitPrice />}
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]:
```suggestion
        {createOrder.type === 'market' ? <ShowMarketPrice /> : <EditLimitPrice />}
```

Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]:
```suggestion
        {createOrder.type === 'market' ? <ShowMarketPrice /> : <EditLimitPrice />}
```

**Review by stackchain — APPROVED** [review]

**Pull request merged**